job_definitions/notify-suppliers-reminder-sign-agreement.yml

```diff
@@ -0,0 +1,73 @@
+- job:
+    name: "remind-suppliers-to-sign-framework-agreement-(43)"
+    display-name: "Remind suppliers to sign framework agreement (43)"
+    project-type: pipeline
+    description: "Send a reminder email to suppliers who have at least one successful lot entry on the given framework but who have not yet signed their framework agreement."
+    parameters:
+      - choice:
+          name: STAGE
+          choices:
+            - preview
+            - staging
+            - production
+      - string:
+          name: FRAMEWORK_SLUG
+          default: null
+          description: "Framework Slug e.g. 'digital-outcomes-and-specialists-5'"
+      - string:
+          name: SUPPLIER_IDS
+          default: null
+          description: "Comma-separated list of Supplier IDs to be emailed (e.g. '1234,5678')"
+      - bool:
+          name: DRY_RUN
+          default: false
+          description: "List notifications that would be sent without sending the emails"
+
+    dsl: |
+      node {
+        try {
+          stage('Clone frameworks') {
+             dir('digitalmarketplace-frameworks') {
+               git url: 'git@github.com:alphagov/digitalmarketplace-frameworks.git', branch: 'master', credentialsId: 'github_com_and_enterprise', poll: true
+             }
+           }
+          stage('Run script') {
+            sh('''
+                if [ -n "$SUPPLIER_IDS" ]; then
+                  FLAGS="$FLAGS --supplier-id=$SUPPLIER_IDS"
+                fi
+
+                if [ "$DRY_RUN" = "true" ]; then
+                  FLAGS="$FLAGS --dry-run"
+                fi
+
+                docker run --rm \
+                -e DM_DATA_API_TOKEN_${STAGE^^} \
```

Comment:
That's really neat - is this a bash thing?

Reply:
yes, apparently this was added in Bash 5 https://www.gnu.org/software/bash/manual/bash.html#index-parameter-expansion described as ``${parameter^^pattern}``